### PR TITLE
Remove unsupported Windows SAC  test images

### PR DESCRIPTION
**Pull request by jsturtevant**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Removes the SAC releases that are EOL from the test images which save space in registry.  

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

/sig windows
/sig testing

/cc @claudiubelu @marosset 
#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by marosset — APPROVED** [review]
/lgtm

/hold
until we remove this job
https://testgrid.k8s.io/sig-windows-containerd-runtime-signal#win-2004-containerd-master-integration

File: test/images/README.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ It can be used by anyone, but if you need to build your own, you can read more a
 
 For Windows, in order to spawn process-isolated containers, the container OS version should closely match
 the host OS version. For this reason, we build test images for different Windows OS Versions: 1809 (Windows Server 2019),
-2004, 20H2, ltsc2022. In order add support for a new Windows OS version, a new entry for that OS version will have
+20H2, ltsc2022. In order add support for a new Windows OS version, a new entry for that OS version will have
```
Comment (acumino) [inline test/images/README.md:98]:
```suggestion
20H2, ltsc2022. In order to add support for a new Windows OS version, a new entry for that OS version will have
```

**Review by acumino — CHANGES_REQUESTED** [review]
Nit

**Comment by claudiubelu** [comment]
It might also be a good idea to bump the ``VERSION`` numbers.

**Comment by jsturtevant** [comment]
@claudiubelu updated the version numbers

**Comment by jsturtevant** [comment]
One of the prow test containers failed to start and the kind job had a flake

```
nit container clonerefs not ready: (state: running) Init container initupload not ready: (state: waiting, reason: "PodInitializing", message: "") Init container place-entrypoint not ready: (state: waiting, reason: "PodInitializing", message: "")
```

/retest

File: test/images/windows-servercore-cache/VERSION [hunk 2]
```diff
@@ -1 +1 @@
-1.0
+1.1
```
Comment (claudiubelu) [inline test/images/windows-servercore-cache/VERSION:1]: we should leave this one at 1.0. This image is never being published, we only need it into staging so we can use to build image image using it.

**Review by claudiubelu — COMMENTED** [review]
One small note. Other than that, LGTM.

File: test/images/windows-servercore-cache/VERSION (hunk 2 above)
Comment (jsturtevant) [reply test/images/windows-servercore-cache/VERSION]: updated, thanks for the explanation

**Pull request merged**